### ISSUE TITLE
Add "float64 only" flag

### ISSUE THREAD

**Issue by axch** [google/jax#4967]
Jax currently has a flag that can be interpreted as "ignore dtype solicitations and run everything in float32 (with warnings)."  Can you add one for forcing float64?  And make them dynamically switchable without exiting the Python process?  This would be very nice for a workflow for debugging numerical accuracy where one runs the same computation in float64 and in float32 and compares the results.

@jakevdp ?

**Comment by jakevdp** [comment]
I don't think we'll add another flag. This kind of modalism in the API is problematic for several reasons, and we're working towards removing it. The current behavior of the `JAX_ENABLE_X64` flag does roughly what you want: everything will default to 64-bit unless you are explicitly setting other precisions.

Regarding making it dynamically switchable within an existing process: that would take a significant amount of redesign of the X64 flag implementation. Because this is behavior we're looking to remove from the package, that sort of investment is probably not warranted.

**Comment by mattjj** [comment]
If setting `JAX_ENABLE_X64` isn't quite enough, then it's also possible one might be able to use [`jax.experimental.callback`](https://github.com/google/jax/blob/master/jax/experimental/callback.py) to set up a simple transformation to do this. See for example the `rewrite` example transformation in that file, which can be used to implement simple single-op rewrites of functions; maybe one can have a rewrite that converts any input/output values to 64bit precision. WDYT?

**Comment by jakevdp** [comment]
I'll note we recently added the `JAX_NUMPY_DTYPE_PROMOTION` flag, which allows you to specify strict dtype promotion for all or part of your code. For example:
```python
with jax.numpy_dtype_promotion('strict'):
  jnp.float32(1) + jnp.float64(1)  # error
```
This I think hits part of the intended usecase in this question.
